http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1303,8 +1303,8 @@ static void recordRequestComplete(RequestStats requestStats, Request request, lo
             return;
         }
 
-        Duration responseProcessingTime = Duration.nanosSince(responseStart);
-        Duration requestProcessingTime = new Duration(responseStart - requestStart, NANOSECONDS);
+        Duration responseProcessingTime = new Duration(max(0, System.nanoTime() - responseStart), NANOSECONDS);
```

Comment:
This also happens in task stats in trino (not just in the airlift client) see: https://github.com/trinodb/trino/issues/11039

If we want to guard against these issues, wouldn't it make more sense to do so in Duration itself/airlift units?